lxd/identities.go

```diff
@@ -293,6 +346,211 @@ func createIdentityTLS(d *Daemon, r *http.Request) response.Response {
+//
+//	Issue a token for a bearer identity.
+//
+//	Issues a new token for the bearer identity and revokes any existing token.
+//
+//	---
+//	consumes:
+//	  - application/json
+//	produces:
+//	  - application/json
+//	parameters:
+//	  - in: body
+//	    name: Token request
+//	    description: Parameters of token creation
+//	    required: true
+//	    schema:
+//	      $ref: "#/definitions/IdentityBearerTokenPost"
+//	responses:
+//	  "200":
+//	    $ref: "#/responses/IdentityBearerToken"
+//	  "400":
+//	    $ref: "#/responses/BadRequest"
+//	  "403":
+//	    $ref: "#/responses/Forbidden"
+//	  "404":
+//	    $ref: "#/responses/NotFound"
+//	  "500":
+//	    $ref: "#/responses/InternalServerError"
+func tokenIdentityIssueToken(d *Daemon, r *http.Request) response.Response {
+	var req api.IdentityBearerTokenPost
+	err := json.NewDecoder(r.Body).Decode(&req)
+	if err != nil {
+		return response.BadRequest(err)
+	}
+
+	s := d.State()
+
+	id, err := request.GetContextValue[*dbCluster.Identity](r.Context(), ctxClusterDBIdentity)
+	if err != nil {
+		return response.SmartError(err)
+	}
+
+	var secret []byte
+	err = s.DB.Cluster.Transaction(r.Context(), func(ctx context.Context, tx *db.ClusterTx) error {
+		secret, err = dbCluster.RotateBearerIdentitySigningKey(ctx, tx.Tx(), id.ID)
+		if err != nil {
+			return err
+		}
+
+		return nil
+	})
+	if err != nil {
+		return response.SmartError(err)
+	}
+
+	token, err := encryption.GetSignedJWT(secret, nil, id.Identifier, s.GlobalConfig.ClusterUUID(), req.Expiry)
+
+	networkCert := s.Endpoints.NetworkCert()
+	serverCert := s.ServerCert()
+	notify := newIdentityNotificationFunc(s, r, networkCert, serverCert)
+	_, err = notify(lifecycle.IdentityUpdated, api.AuthenticationMethodBearer, id.Identifier, true)
+	if err != nil {
+		return response.SmartError(err)
+	}
+
+	return response.SyncResponse(true, api.IdentityBearerToken{Token: token})
+}
+
+// swagger:operation POST /1.0/auth/identities/bearer/{nameOrID}/token identities identity_delete_bearer_token
+//
+//	Revoke a bearer identity token.
+//
+//	Revokes any existing token for the identity.
+//
+//	---
+//	consumes:
+//	  - application/json
+//	produces:
+//	  - application/json
+//	responses:
+//	  "403":
+//	    $ref: "#/responses/Forbidden"
+//	  "404":
+//	    $ref: "#/responses/NotFound"
+//	  "500":
+//	    $ref: "#/responses/InternalServerError"
+func tokenIdentityRevokeToken(d *Daemon, r *http.Request) response.Response {
+	s := d.State()
+
+	id, err := request.GetContextValue[*dbCluster.Identity](r.Context(), ctxClusterDBIdentity)
+	if err != nil {
+		return response.SmartError(err)
+	}
+
+	err = s.DB.Cluster.Transaction(r.Context(), func(ctx context.Context, tx *db.ClusterTx) error {
+		rowsAffected, err := dbCluster.DeleteBearerIdentitySigningKey(ctx, tx.Tx(), id.ID)
+		if err != nil {
+			return fmt.Errorf("Failed to revoke token: %w", err)
+		}
+
+		if rowsAffected == 0 {
```

Comment:
Yeah this would be better handled inside DeleteBearerIdentitySigningKey in my view.